capstone/src/test/java/com/google/sps/servlets/ExploreServletTest.java

```diff
@@ -0,0 +1,129 @@
+import javax.servlet.http.HttpServletResponse;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+/**
+ * Class to test ExploreServlet by adding a few mock clubs and ensuring their existence and
+ * properties
+ */
+@RunWith(JUnit4.class)
+public final class ExploreServletTest {
+
+  private static final String KEVIN = "kshao@google.com";
+  private static final String MEGHA = "kakm@google.com";
+  private static final String MEGAN = "meganshi@google.com";
+
+  private static final String CLUB_1 = "Club 1";
+  private static final String CLUB_2 = "Club 2";
+  private static final String CLUB_3 = "Club 3";
+
+  private static final String SITE_1 = "www.fakesite.com";
+  private static final String SITE_2 = "www.realfakesite.com";
+
+  private static final String DESCRIPTION_1 = "Helping people.";
+  private static final String DESCRIPTION_2 = "Not helping people.";
+
+  private ExploreServlet servlet;
+  @Mock private HttpServletRequest request;
+  @Mock private HttpServletResponse response;
+  private DatastoreService datastore;
+
+  private LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(
+          new LocalDatastoreServiceTestConfig(),
+          new LocalUserServiceTestConfig(),
+          new LocalURLFetchServiceTestConfig());
+
+  @Before
+  public void setUp() {
+    this.servlet = new ExploreServlet();
+    MockitoAnnotations.initMocks(this);
+    helper.setUp();
+    this.datastore = DatastoreServiceFactory.getDatastoreService();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Test
+  public void correctNumberReturned() throws IOException {
+    helper.setEnvEmail("kshao").setEnvAuthDomain("gmail.com").setEnvIsLoggedIn(true);
+    when(request.getParameter(Constants.CLUB_NAME_PROP)).thenReturn(CLUB_2);
+
+    Entity club1 = new Entity(Constants.CLUB_PROP);
+    club1.setProperty(Constants.CLUB_NAME_PROP, CLUB_1);
+    club1.setProperty(Constants.MEMBER_PROP, ImmutableList.of(MEGHA, MEGAN, KEVIN));
+    club1.setProperty(Constants.OFFICER_PROP, ImmutableList.of(MEGHA));
+    club1.setProperty(Constants.DESCRIP_PROP, DESCRIPTION_1);
+    club1.setProperty(Constants.WEBSITE_PROP, SITE_1);
+
+    Entity club2 = new Entity(Constants.CLUB_PROP);
+    club2.setProperty(Constants.CLUB_NAME_PROP, CLUB_2);
+    club2.setProperty(Constants.MEMBER_PROP, ImmutableList.of(KEVIN));
+    club2.setProperty(Constants.OFFICER_PROP, ImmutableList.of(KEVIN));
+    club2.setProperty(Constants.DESCRIP_PROP, DESCRIPTION_2);
+    club2.setProperty(Constants.WEBSITE_PROP, SITE_2);
+
+    this.datastore.put(club1);
+    this.datastore.put(club2);
+
+    JsonArray response = getServletResponse(servlet);
+
+    int expectedSize = 2;
+    Assert.assertEquals(expectedSize, response.size());
```

Comment:
Let's test more than just the size assertion. We should also test the individual objects returned to ensure they have the fields we expect.